cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -1483,10 +1521,14 @@ def attach_subnet(self, subnet):
             subnet = self._provider.networking.subnets.get(subnet)
         if subnet.network_id == self.network_id:
             return
-        cb.log.warning('GCE routers should be attached at creation time')
+        cb.log.warning('Google Cloud Routers automatically learn new subnets '
+                       'in your VPC network and announces them to your '
+                       'on-premises network')
 
     def detach_subnet(self, network_id):
-        cb.log.warning('GCE routers are always attached')
+        cb.log.warning('Cannot detach from subnet. Google Cloud Routers '
+                       'automatically learn new subnets in your VPC network '
+                       'and announces them to your on-premises network')
```

Comment:
Does this mean that there's no way to isolate two subnets in GCE? All subnets will automatically start communicating with each the moment a router is present?

Reply:
As far as I understand it, GCP routers are for controlling the traffic between your GCP network and your on-premises networks, not for controlling traffic among subnets in your GCP network.

From the interface itself, it's not clear to me what is intended by router.{attach/detach}_subnet methods.

If you want to isolate some subnets from on-premises clients, we can use the custom mode of the GCP routers and advertise specific subnets explicitly instead of relying on the automatic detection of subnets, which is the default behaviour (https://cloud.google.com/router/docs/how-to/advertising-subnets). Should I do this?

@mbookman for opinions and verifying what I said.

Reply:
The intent of router.attach_subnet is to allow routing between subnets as well as to route external gateway traffic to the subnet. We definitely haven't considered routing between on-premise and VPC yet, so this is more about subnet to subnet comms.

Reply:
It looks like instead of creating a Google Cloud Router, the GCE implementation here should be updating the routing table that is associated with the VPC:

https://cloud.google.com/vpc/docs/routes

I *think* this is similar to the AWS implementation, if I'm reading it correctly.

Reply:
@nuwang : can you point to an explanation of the use case for the routing between subnets and the internet and providing end user control? Want to make sure that the implementation is matches the expected semantics.

Also - I believe this is the last issue currently for this PR. It would be great if we could come up with a solution to get this PR merged.

Reply:
@mbookman Sorry, I misspoke - the main requirement behind the router was subnet<->internet comms, not subnet<->subnet comms. We are currently not using any kind of subnet<->subnet comms and with Kubernetes, don't foresee an immediate need for it. Nevertheless, conceptually, if all subnets connected to the same router can route between each other, that would be desirable, but not critical.